avro-converter/src/main/java/io/confluent/connect/avro/AvroData.java

```diff
@@ -590,7 +628,26 @@ private static Object maybeWrapSchemaless(Schema schema, Object value, String ty
         baseSchema = org.apache.avro.SchemaBuilder.builder().booleanType();
         break;
       case STRING:
-        baseSchema = org.apache.avro.SchemaBuilder.builder().stringType();
+        if (schema.parameters() != null && schema.parameters().containsKey(AVRO_TYPE_ENUM)){
+
+          String enumSchemaName = schema.parameters().get(AVRO_TYPE_ENUM);
```

Comment:
correct the new Avro Type Enum is new, as such historic stuff won't contain this and will be still handled like a string.